packages/python/plotly/plotly/express/_core.py

```diff
@@ -754,6 +754,90 @@ def apply_default_cascade(args):
         args["marginal_x"] = None
 
 
+def build_or_augment_dataframe(args, attrables, array_attrables, constructor):
+    """
+    Constructs an implicit dataframe and modifies `args` in-place.
+    `attrables` is a list of keys into `args`, all of whose corresponding
+    values are converted into columns of a dataframe.
+    Used to be support calls to plotting function that elide a dataframe
+    argument; for example `scatter(x=[1,2], y=[3,4])`.
+    """
+
+    # We start from an empty DataFrame except for the case of functions which
+    # implicitely need all dimensions: Splom, Parcats, Parcoords
+    # This could be refined when dimensions is given
+    if constructor in [go.Splom, go.Parcats, go.Parcoords]:  # we take all dimensions
+        df = args["data_frame"]
```

Comment:
even in this case I would not directly use the provided data frame: rather I would start with an empty `df` always, and move the default `dimensions` logic here (basically if `args["dimensions"]` is present but empty then add all the columns from `args["data_frame"]` to `df`)... using this logic would also enable you to not pass `constructor` in here. (In general, the less we can condition things on `constructor` the better IMO :)